packages/debug-test/CDPDebugger.ts

```diff
@@ -0,0 +1,400 @@
+		columnNumber: number,
+		scriptHash: string,
+		condition: string = "") : Promise<BreakpointInfo> {
+		const commandId = this.nextCommandId++;
+		
+		// the actual CDP message
+		// TODO: pass on urlRegex. scriptHash
+		const command = `{"id":${commandId}, "method":"Debugger.setBreakpointByUrl", "params":{
+			"url":${url},
+			"lineNumber":${lineNumber},
+			"columnNumber":${columnNumber},
+			"condition": ${condition}
+		}}`;
+
+		// a short description for logging
+		const commandDescription = `Debugger.setBreakpointByUrl (id: ${commandId})`;
+
+		const expectedResponse1 = {"id": commandId, "result": {}};
+
+		const resultPromise = new Promise<BreakpointInfo>((resolve, reject) => {
+			this.expectedResponses.push([
+				`result response for ${commandDescription}`,
+				response => {
+					if (response.hasOwnProperty("id") && response.id == commandId) {
+						resolve(response.result);
+						return true;
+					}
+					return false;
+				}
+			]);
+		});
+
+		await this.wsOpened;
+		testLog.message(`sending ${commandDescription}`);
+		this.ws.send(command);
+		this.lastCommand = commandDescription;
+		return resultPromise;
+	}
+
+	/**
+	 * Sends a Debugger.removeBreakpoint CDP message (see https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-removeBreakpoint).
+	 * @param breakpointId Breakpoint identifier (e.g. result of DebuggerSetBreakpointByUrl)
+	 */
+	public async DebuggerRemoveBreakpoint(breakpointId: string): Promise<void> {
+		const commandId = this.nextCommandId++;
+		
+		// the actual CDP message
+		const command = `{"id":${commandId}, "method":"Debugger.removeBreakpoint", "params":{"breakpointId":"${breakpointId}"}}`;
+
+		// a short description for logging
+		const commandDescription = `Debugger.removeBreakpoint (id: ${commandId})`;
+
+		const expectedResponse1 = {"id": commandId, "result": {}};
+
+		const resultPromise = new Promise<void>((resolve, reject) => {
+			this.expectedResponses.push([
+				`result response for ${commandDescription}`,
+				actualResponse => {
+					// relying on object comparison to be independent of key order differences
+					if (lodash.isEqual(expectedResponse1, actualResponse)) {
+						resolve();
+						return true;
+					}
+					return false;
+				}
+			]);
+		});
+
+		await this.wsOpened;
+		testLog.message(`sending ${commandDescription}`);
+		this.ws.send(command);
+		this.lastCommand = commandDescription;
+		return resultPromise;
+	}
+
+	public async checkOutstandingResponses(timeout: number) {
+		return new Promise<void>(async (resolve, reject) => {
+			const pollPeriod = 500;
+			const retryCount =  Math.ceil(timeout / pollPeriod);
+			const startTime = new Date();
+
+			let i = 0;
+			do {
+				if (this.expectedResponses.length == 0) {
+					testLog.message("received all expected responses");
+					resolve();
+					return;
+				}
+
+				if ((new Date()).getTime() - startTime.getTime() >= timeout) break;
+
+				await sleep(pollPeriod);
+			 } while (++i < retryCount);
```

Comment:
`app.WaitUntil()` tries to achieve a similar goal, or repeatedly polling for a condition. Can it be used here as well?